test_vm/src/lib.rs

```diff
@@ -82,15 +83,15 @@ impl<'bs> TestVM<'bs> {
         }
     }
 
-    pub fn new_with_singletons(store: &'bs MemoryBlockstore) -> TestVM<'bs> {
+    pub fn new_with_singletons(store: Rc<MemoryBlockstore>) -> TestVM {
```

Comment:
I'd consider defining this as `store: impl Into<Rc<MemoryBlockstoire>>`, allowing the user to pass either a `MemoryBlockstore` or an `Rc<MemoryBlockstore>>`. I.e.

```suggestion
    pub fn new_with_singletons(store: impl Into<Rc<MemoryBlockstore>>) -> TestVM {
        let store = store.into();
```

Same with `TestVM::new`.

But let's wait for @anorth to comment on this before continuing.

Reply:
This would let us call `TestVM::new_with_singletons(store)` without having to explicitly wrap in an `Rc`.

Reply:
Yes please (for `new` too)

Reply:
@Stebalien Where is the implementation for 

```
impl Into<Rc<MemoryBlockstore>> for MemoryBlockstore {
}
```

defined though in the code ? I also don't see a 

```
impl From<MemoryBlockstore> for Rc<MemoryBlockstore>
```

anywhere. Does the Rust stdlib have come default impl here to convert a MemoryBlockstore to an Rc<MemoryBlockstore> ?

Reply:
First, there's also a blanket implementation of `Into<U> for T where U: From<T>`. That is, you _always_ implement `From<T> for U` and get an automatic implementation of `Into<U> for T`. Never implement `Into` directly. I recommend reading the [`Into`](https://doc.rust-lang.org/std/convert/trait.Into.html) documentation and the [`From` and `Into`](https://doc.rust-lang.org/rust-by-example/conversion/from_into.html) section of the book.

Additionally, there's a blanket implementation of [`From<T> for Rc<T>`](https://doc.rust-lang.org/std/rc/struct.Rc.html#impl-From%3CT%3E-for-Rc%3CT,+Global%3E).

Reply:
Thanks @Stebalien ! Yeah the blanket implementation of `From<T> for Rc<T>` is what I missing on.